tests/test_registration_and_login.py

```
@@ -0,0 +1,180 @@
+import os
+
+import pytest
+from dotenv import load_dotenv
+
+# Import the application and database setup
+from hushline import create_app, db
+from hushline.generate_invite_codes import create_invite_code
+
+# Import models and other modules
+from hushline.model import User
+
+
+@pytest.fixture
+def client():
+    # Load environment variables from .env file or the equivalent
+    load_dotenv("env.sh")
+
+    # Use a simple file name without special characters
+    db_file = "hushline.db"
+
+    # Construct the database path based on the environment
+    if os.environ.get("GITHUB_ACTIONS"):
+        # GitHub Actions environment
+        db_dir = "/home/runner/work/hushline/hushline"
+        os.makedirs(db_dir, exist_ok=True)
+    else:
+        # Local testing environment
+        db_dir = "/tmp/hushline"
+
+    db_path = os.path.join(db_dir, db_file)
+
+    print(f"SQLite database path: {db_path}")  # Print the database path for debugging
+
+    app = create_app()
+    app.config["TESTING"] = True
+    app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///:memory:"
+    app.config["WTF_CSRF_ENABLED"] = False
+    app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+
+    print("Creating test client...")  # Debug statement
+
+    # Setup the app context and database schema
+    with app.app_context():
+        print("Creating database tables...")  # Debug statement
+        try:
```

don't `try` here. just let the exception raise